.github/workflows/bench.yml

```diff
@@ -86,24 +86,29 @@ jobs:
         echo "INDEX_STATE=$INDEX_STATE1" >> $GITHUB_ENV
 
     - name: Form the package list ('cabal.project.freeze')
-      continue-on-error: true
+      id: compute-cache-key
       run: |
         cabal v2-freeze && \
         echo '' && \
         echo 'Output:' && \
         echo '' && \
         cat 'cabal.project.freeze' && \
         echo '' || \
-        echo 'WARNING: Could not produce the `freeze`.
+        echo 'WARNING: Could not produce the `freeze`.'
+        echo ::set-output name=value::${{ hashFiles('cabal.project.freeze') }}
+        # Removing freeze file as it can break builds using allow-newer
+        rm -f cabal.project.freeze
 
     - name: Hackage sources cache
       uses: actions/cache@v2
       env:
         cache-name: hackage-sources
       with:
-        path: ${{ env.CABAL_PKGS_DIR }}
-        key:          ${{ env.cache-name }}-${{ env.INDEX_STATE }}
-        restore-keys: ${{ env.cache-name }}-
+        path:  ${{ env.CABAL_PKGS_DIR }}
+        key:   ${{ env.cache-name }}-${{ env.INDEX_STATE }}-${{ steps.compute-cache-key.outputs.value }}
+        restore-keys: |
+               ${{ env.cache-name }}-${{ env.INDEX_STATE }}-
+               ${{ env.cache-name }}-
```

Comment:
Please, keep the old version.

Skimming from full 100% caching of sources - is where we can save a lot of space.

The size of the sources ~ equals the size of the build results.

If we include `freeze` distinction on sources - that would produce x15 times more copies of source cache without real benefit.

Sources are fast to download. CI can live even without caching of sources.
Cachin of sources is done only for the purpose of the `cashing` workflow & reused only inside of it. The 30%-0% of sources that need additional download between versions in the matrix - downloading them takes seconds.

And `caching` build is an asynchronous process to the PR cycle, so downloading sources for a couple of seconds does not really affect PR loop speed.

That is why `key` is kept such - `index-state` is considered enough of a metric to be roughly in the area all builds need  (or how much `index-state` they have).

Reply:
sure, I assumed the difference between caching and the others was not wanted

Reply:
When CI works properly - other `cabal` workflows should not need to download source code ever. Except when PR diff actively changes package configuration.

This is source code caching. It is used only inside the `caching` workflow. The workflow builds those sources & so what gets reused between workflows is compiled results.

We keep only 1 storage space (that depends on `index-state`) of it & in that way share most of the sources between `caching` jobs while at the same time not a lot of caching pool space.

Reply:
Oh, yes. Sorry. `hackage-sources` is still provided to other workflows, but main cause/use of it is for `caching` & to have minimal yet satisfactory source sharing.